accepted/future-releases/inline-classes/feature-specification.md

```diff
@@ -0,0 +1,969 @@
+there is no ambiguity.
+
+*For example:*
+
+```dart
+inline class V2 {
+  final Object id;
+  V2(this.id);
+  void foo() { print('V2.foo()'); }
+}
+
+inline class V3 {
+  final Object id;
+  V3(this.id);
+  void foo() { print('V3.foo()'); }
+}
+
+inline class V1 implements V2, V3 {
+  final Object id;
+  V1(this.id);
+  void bar() {
+    super.V3.foo(); // Prints "V3.foo()".
+  }
+}
+```
+
+
+## Dynamic Semantics of Inline Classes
+
+The dynamic semantics of inline member invocation follows from the code
+transformation specified in the section about the static analysis.
+
+*In short, with `e` of type
+<code>Inline&lt;T<sub>1</sub>, .. T<sub>s</sub>&gt;</code>,
+`e.m(args)` is treated as
+<code>invokeInlineMethod(Inline, &lt;T<sub>1</sub>, .. T<sub>s</sub>&gt;, e).m(args)</code>.
+Similarly for getters, setters, and operators.*
+
+Consider an inline class declaration _DV_ named `Inline` with
+representation name `id` and representation type `R`.  Invocation of a
+non-redirecting generative inline class constructor proceeds as follows: A
+fresh, non-late, final variable `v` is created. An initializing formal
+`this.id` has the side-effect that it initializes `v` to the actual
+argument passed to this formal. An initializer list element of the
+form `id = e` or `this.id = e` is evaluated by evaluating `e` to an
+object `o` and binding `v` to `o`.  During the execution of the
+constructor body, `this` and `id` are bound to the value of `v`.  The
+value of the instance creation expression that gave rise to this
+constructor execution is the value of `this`.
+
+At run time, for a given instance `o` typed as an inline type `V`, there
+is _no_ reification of `V` associated with `o`.
+
+*This means that, at run time, an object never "knows" that it is being
+viewed as having an inline type. By soundness, the run-time type of `o`
+will be a subtype of the representation type of `V`.*
+
+The run-time representation of a type argument which is an inline type
+`V` is the corresponding instantiated representation type.
+
+*This means that an inline type and the underlying representation type
+are considered as being the same type at run time. So we can freely
+use a cast to introduce or discard the inline type, as the static type
+of an instance, or as a type argument in the static type of a data
+structure or function involving the inline type.*
+
+A type test, `o is U` or `o is! U`, and a type cast, `o as U`, where `U` is
+or contains an inline type, is performed at run time as a type test and type
+cast on the run-time representation of the inline type as described above.
+
+
+### Summary of Typing Relationships
+
+*Here is an overview of the subtype relationships of an inline type
+`V0` with representation type `R` and superinterfaces `V1 .. Vk`, as
+well as other typing relationships involving `V0`:*
+
+- *`V0` is a subtype of `Object?`.*
+- *`V0` is a supertype of `Never`.*
+- *If `R` is a top type then `V0` is a top type,
```

Comment:
I don't follow this.  Why would `V0` be a top type here?  I would not expect that given:

```dart
inline class V0 {
  final Object? x;
  V0(this.x);
}
```

that `String` would be a subtype of `V0`.

Reply:
Agree. Being a subtype of `Object?` does not make it a *supertype* of `Object?`.

This text may be a hold-over from when the views specification made the representation type a subtype of the view type?

Reply:
Exactly, that bit came from earlier versions. Removed it.